src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -80,6 +80,7 @@
      * dynamically discoverable.
      * @return false if the class shouldn't be used; otherwise true
      */
+    // TODO: I don't think that this should be bounded, but I think that the default value should return c -> c instanceof getPluginClass()
     public Predicate<Class<?>> getClassFilter() { return c -> true;}
```

Comment:
If this is a public method that some client may use, I think that it should not be bounded but at least being:
```java
public Predicate<Class<?>> getClassFilter() {
    c -> c.isAssignableFrom(getPluginClass())
}
```
Or something similar.